src/algokit/core/sandbox.py

```diff
@@ -120,6 +121,49 @@ def ps(self, service_name: str | None = None) -> list[dict[str, Any]]:
         assert isinstance(data, list)
         return cast(list[dict[str, Any]], data)
 
+    def _get_local_image_version(self, image_name: str) -> str:
+        """
+        Get the local version of a Docker image
+        """
+        arg = "{{.RepoDigests}}"
+        local_version = run(
+            ["docker", "image", "inspect", image_name, "--format", arg],
+            cwd=self.directory,
+            bad_return_code_error_message="Failed to get image inspect",
+        )
+        # Remove brackets and split on '@' symbol and get the SHA hash
+        return local_version.output[1:-1].split("@")[1]
+
+    def _get_latest_image_version(self, image_name: str) -> str:
+        """
+        Get the latest version of a Docker image from Docker Hub
+        """
+        args = image_name.split(":")
+        name = args[0]
+        tag = args[1] if len(args) > 1 else "latest"
+        url = f"https://registry.hub.docker.com/v2/repositories/{name}/tags/{tag}"
+        try:
+            data = httpx.get(url=url)
+            return str(data.json()["digest"])
+        except Exception as err:
+            logger.debug(f"Error checking indexer status: {err}", exc_info=True)
+            return ""
+
+    def check_docker_compose_for_new_image_versions(self) -> None:
+        """
+        Check Docker Compose file for new image versions
+        """
+        docker_compose = yaml.safe_load(get_docker_compose_yml())
```

Comment:
Rather than loading the docker-compose.yaml contents, could instead make the algod and indexer image tags constants and reference them here and in the `get_docker_compose_yml`